Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -98,10 +98,9 @@ private void TrackSplitsCost(List<Split> splits)
 
     async Task DoWork(Split split, CancellationToken cancellationToken)
     {
-      var checker = await split.parent.CheckerPool.FindCheckerFor(split.parent, split);
+      var checker = await split.parent.CheckerPool.FindCheckerFor(split.parent, split, cancellationToken).WaitAsync(cancellationToken);
```

Comment:
Why did you add `.WaitAsync(cancellationToken)`? I think that's only useful when the Task doesn't have cancellation support, but that's what you're already adding in this PR